projects/js-packages/publicize-components/src/components/form/index.js

```diff
@@ -22,41 +25,81 @@ import styles from './styles.module.scss';
+ * @param {number} props.numberOfSharesRemaining        - The number of shares remaining for the current period. Optional.
  * @returns {object}                                    - Publicize form component.
  */
 export default function PublicizeForm( {
 	isPublicizeEnabled,
 	isRePublicizeFeatureEnabled,
 	isPublicizeDisabledBySitePlan,
+	numberOfSharesRemaining = null,
 } ) {
-	const { connections, toggleById, hasConnections } = useSocialMediaConnections();
+	const {
+		connections,
+		toggleById,
+		hasConnections,
+		enabledConnections,
+	} = useSocialMediaConnections();
 	const { message, updateMessage, maxLength } = useSocialMediaMessage();
 
 	const isDisabled = () =>
 		! isRePublicizeFeatureEnabled && connections.every( connection => ! connection.toggleable );
 	const Wrapper = isPublicizeDisabledBySitePlan ? Disabled : Fragment;
 
+	const outOfConnections =
+		numberOfSharesRemaining !== null && numberOfSharesRemaining <= enabledConnections.length;
+
 	return (
 		<Wrapper>
 			{ hasConnections && (
-				<PanelRow>
-					<ul className={ styles[ 'connections-list' ] }>
-						{ connections.map(
-							( { display_name, enabled, id, service_name, toggleable, profile_picture } ) => (
-								<PublicizeConnection
-									disabled={ isRePublicizeFeatureEnabled ? ! isPublicizeEnabled : ! toggleable }
-									enabled={ enabled && ! isPublicizeDisabledBySitePlan }
-									key={ id }
-									id={ id }
-									label={ display_name }
-									name={ service_name }
-									toggleConnection={ toggleById }
-									profilePicture={ profile_picture }
-								/>
-							)
-						) }
-					</ul>
-				</PanelRow>
+				<>
+					{ ! isDisabled() && numberOfSharesRemaining !== null && (
+						<PanelRow>
+							<Notice type={ numberOfSharesRemaining < connections.length ? 'warning' : 'default' }>
+								{ createInterpolateElement(
+									sprintf(
+										/* translators: %d is the number of shares remaining, upgradeLink is the link to upgrade to a different plan */
+										_n(
+											'You have %d share remaining. <upgradeLink>Upgrade</upgradeLink> to share to all your social media accounts.',
+											'You have %d shares remaining. <upgradeLink>Upgrade</upgradeLink> to share to all your social media accounts.',
+											numberOfSharesRemaining,
+											'jetpack'
+										),
+										numberOfSharesRemaining
+									),
+									{
+										upgradeLink: (
+											<ExternalLink
+												href={ getRedirectUrl( 'jetpack-social-basic-plan-block-editor' ) }
+											/>
+										),
+									}
+								) }
+							</Notice>
+						</PanelRow>
+					) }
+					<PanelRow>
+						<ul className={ styles[ 'connections-list' ] }>
+							{ connections.map(
+								( { display_name, enabled, id, service_name, toggleable, profile_picture } ) => (
+									<PublicizeConnection
+										disabled={
+											( isRePublicizeFeatureEnabled ? ! isPublicizeEnabled : ! toggleable ) ||
```

Comment:
This looks like the plan checks that we want to remove for Republicize. Should we do that now?

Reply:
@spsiddarthan is working on a PR to remove these, I think it would be good not to touch that logic in this PR and instead bring them together later.

Reply:
Ah of course. I forgot that these checks were actually being used currently!